lib/database_transform/schema_table.rb

```diff
@@ -131,8 +134,24 @@ def time_block(before, after, &proc)
   def transform!(schema)
     # For each item in the old model
     default_scope = @default_scope || @source.method(:all)
-    @source.instance_exec(&default_scope).each do |record|
-      transform_record!(schema, record)
+    collection = @source.instance_exec(&default_scope)
+
+    transform_in_batches!(collection) do |group|
+      group.each { |record| transform_record!(schema, record) }
+    end
+  end
+
+  # Transform a collection of records in batches. This method uses `find_in_batches` to split the large collection and
+  # improve performance.
+  #
+  # @param [Array|ActiveRecord_Relation] collection The records.
+  # @param [Proc] block The block to transform a group of records.
+  # @return [Void]
+  def transform_in_batches!(collection, &block)
```

Comment:
Actually, I think this is a more generic function, maybe call it `in_batches(collection, &block)`? Because the transform is really the block given by `#transform!`